scripts/LOffice.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/python3
+
+import uno
+import time
+import os
+import logging
+
+logging.basicConfig(format='%(name)s - %(levelname)s - %(message)s', level=logging.INFO)
+logger = logging.getLogger(os.path.basename(__file__))
+
+autosave_time = float(os.environ.get('AUTOSAVE_TIME', 60))
+
+def connect():
+
+    while True:
+        try:
+            logger.debug("Establish connection with LO API")
+            localContext = uno.getComponentContext()
+            logger.debug("local context ready")
+            resolver = localContext.ServiceManager.createInstanceWithContext("com.sun.star.bridge.UnoUrlResolver", localContext)
+            logger.debug("resolver ready")
+            ctx = resolver.resolve("uno:pipe,name=open365_LO;urp;StarOffice.ComponentContext")
+            logger.debug("Component context ready")
+            logger.info('Connected to libreoffice-uno')
+            smgr = ctx.ServiceManager
+            # Get the Libreoffice Desktop
+            desktop = smgr.createInstanceWithContext("com.sun.star.frame.Desktop", ctx)
+            logger.debug("Desktop ready")
+
+            return desktop
+
+        except Exception:
+            logger.warning("Could not connect to LibreOffice")
+            time.sleep(1)
+
+def saveAllDocs(desktop):
```

Comment:
This function can crash.

The connection to LO could be invalidated for a small amount of time and then when we try and use this connection, both LibreOffice and this script can crash. It's hard to reproduce, since it requires things to line up perfectly.

Could you please refactor the code so that these functions always first connect, then perform the action and then disconnect. Making a connection should be cheap given that we're using a pipe.